tests/utils_tests/test_text.py

```diff
@@ -95,17 +95,97 @@ def test_truncate_chars(self):
         )
 
     def test_truncate_chars_html(self):
-        perf_test_values = [
-            (("</a" + "\t" * 50000) + "//>", None),
-            ("&" * 50000, "&" * 9 + "…"),
-            ("_X<<<<<<<<<<<>", None),
-        ]
-        for value, expected in perf_test_values:
-            with self.subTest(value=value):
-                truncator = text.Truncator(value)
-                self.assertEqual(
-                    expected if expected else value, truncator.chars(10, html=True)
-                )
+        truncator = text.Truncator(
+            '<p id="par"><strong><em>The quick brown fox jumped over the lazy dog.</em>'
+            "</strong></p>"
+        )
+        self.assertEqual(
+            '<p id="par"><strong><em>The quick brown fox jumped over the lazy dog.</em>'
+            "</strong></p>",
+            truncator.chars(80, html=True),
+        )
+        self.assertEqual(
+            '<p id="par"><strong><em>'
+            "The quick brown fox jumped over the lazy dog.</em></strong></p>",
+            truncator.chars(46, html=True),
+        )
+        self.assertEqual(
+            '<p id="par"><strong><em>The quick brown fox jumped over the lazy dog…</em>'
+            "</strong></p>",
+            truncator.chars(45, html=True),
+        )
+        self.assertEqual(
+            '<p id="par"><strong><em>The quick…</em></strong></p>',
+            truncator.chars(10, html=True),
+        )
+        self.assertEqual(
+            "…",
+            truncator.chars(1, html=True),
+        )
+        self.assertEqual(
+            "…",
+            truncator.chars(0, html=True),
+        )
```

Comment:
This seems wrong and unexpected. If we're requesting a length of zero, we should have an empty string.

Reply:
I don't disagree! However, this is part of the commit which is adding tests to the current behavior. These differences made working on this more complicated!

If we're happy with a breaking change then we can simplify as you suggested.

Reply:
So I think we add the tests for the current behaviour and then intentionally break it in the main commit to return empty string which makes more sense.

Reply:
I'd still move it to the last commit. We're not changing here a behavior that can be expected by anyone, this is a regression test that should be a part of the bugfix.

Reply:
Can I check my understanding of this? Are you saying to squash this PR to a single commit? (Happy to do so, just don't want to split it out again later...)

Reply:
@felixxm could you please clarify what you mean? David and I understand from your comment that the two commits would be better off squashed into one, is that your ask?

Reply:
> @felixxm could you please clarify what you mean? David and I understand from your comment that the two commits would be better off squashed into one, is that your ask?

No, I'm talking only about the following assertion :
```
        self.assertEqual(
            "…",
            truncator.chars(0, html=True),
        )
```
It should not be a part of the first commit, it's a regression test that should be added as:
```
        self.assertEqual(
            "",
            truncator.chars(0, html=True),
        )
```
in the second commit (with the main change).

Reply:
Thanks for explaining. I've made this change.